src/rust/engine/src/scheduler.rs

```diff
@@ -331,6 +336,23 @@ impl Scheduler {
     })
   }
 
+  async fn wait_for_tail_tasks(tasks: Vec<BoxFuture<'static, ()>>) {
+    if !tasks.is_empty() {
+      log::trace!("waiting for {} tail tasks to complete", tasks.len());
+
+      let joined_tail_tasks_fut = futures::future::join_all(tasks);
+      let timeout_fut = time::timeout(Duration::from_secs(5), joined_tail_tasks_fut);
+      match timeout_fut.await {
+        Ok(_) => {
+          log::trace!("tail tasks completed successfully");
+        }
+        Err(_) => {
+          log::trace!("tail tasks failed to complete within timeout");
```

Comment:
Maybe warn?

Reply:
Possibly. I have another comment suggesting adding names for the tasks. A warning here would be better for the user if it identified what Pants was doing that timed out.